File: src/main/java/quebec/virtualite/backend/services/domain/DomainService.java
package quebec.virtualite.backend.services.domain;

import quebec.virtualite.backend.services.domain.entities.CityEntity;

import java.util.List;
import java.util.Optional;

public interface DomainService
{
    void addCity(CityEntity city);

    void deleteAll();

    void deleteCity(CityEntity city);

    List<CityEntity> getCities();

    Optional<CityEntity> getCity(String cityName);

    void updateCity(CityEntity city);
}

File: src/main/kotlin/quebec/virtualite/backend/services/domain/DomainService.kt
package quebec.virtualite.backend.services.domain

import quebec.virtualite.backend.services.domain.entities.CityEntity

interface DomainService
{
    fun addCity(city: CityEntity)
    fun deleteAll()
    fun deleteCity(name: String)
    fun getCityDetails(city: String): CityEntity?
    fun getCitiesDetails(): List<CityEntity>
    fun updateCity(city: CityEntity)
}